baseplate.go

```diff
@@ -209,7 +209,12 @@ func Serve(ctx context.Context, args ServeArgs) error {
 			select {
 			case <-ctx.Done():
 				// The context timed-out or was cancelled so use that error.
-				err = fmt.Errorf("baseplate: context cancelled while waiting for server.Close(). %w", ctx.Err())
+				switch ctx.Err() {
+				case context.DeadlineExceeded:
+					err = fmt.Errorf("baseplate: context cancelled with timeout while waiting for server.Close(). %w", ctx.Err())
+				default:
+					err = fmt.Errorf("baseplate: context cancelled externally. %w", ctx.Err())
```

Comment:
nit, the original wording seemed fine for the default case
```suggestion
					err = fmt.Errorf("baseplate: context cancelled while waiting for server.Close(). %w", ctx.Err())
```

Reply:
Actually, the initial "while waiting for server.Close()" [seems like] meant that the exception was thrown, because of closing operation was happening too long. The runtimebp.HandleShutdown function notified ShutdownHandler only if runtimebp.HandleShutdown catch os.Signal - this let's ShutdownHandler write to the chanel, which runtimebp.HandleShutdown is waiting for to return. That's why the PR firstly handle the ctx.Done() instead of ignoring. The granulation of exception is minor in this case - just to check actual reason of stoping timeout|deadline